### PR TITLE
Initiate Apostrophe CMS typing

### PR DESCRIPTION
**Pull request by afholderman**
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: types/apostrophe/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,302 @@
+// Type definitions for apostrophe 2.67
+// Project: https://github.com/apostrophecms/apostrophe#readme
+// Definitions by: My Self <https://github.com/me>
```
Comment (amcasey) [inline types/apostrophe/index.d.ts]: Please update the author list.

File: types/apostrophe/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
```
Comment (amcasey) [inline types/apostrophe/index.d.ts]: The lower the TS version number is, the easier the typings are to consume.  At a glance, nothing in these definitions jumped out as requiring TS 2.8 - did I miss something?

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by afholderman** [comment]
@amcasey requested changes have been made.

File: types/apostrophe/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+declare function apostrophe(options: any, ...args: any[]): any;
+
+declare namespace apostrophe {
+    const moogBundle: {
+        directory: string;
+        modules: string[];
+    };
+
+    const prototype: {};
```
Comment (amcasey) [inline types/apostrophe/index.d.ts]: Out of curiosity, why would a namespace have a `prototype` property?

File: types/apostrophe/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        | "apostrophe-urls"
+        | "apostrophe-users"
+        | "apostrophe-utils"
+        | "apostrophe-versions"
+        | "apostrophe-video-fields"
+        | "apostrophe-video-widgets"
+        | "apostrophe-widgets";
+
+    // Pass in custom modules to AposModuleOptions to allow them in extend
+    interface AposModuleOptions<C> {
```
Comment (amcasey) [inline types/apostrophe/index.d.ts]: Consider making this type parameter optional (i.e. `C = {}`).

**Comment by amcasey** [comment]
@afholderman, Thanks!  I added a couple more comments, but they're not urgent.

File: types/apostrophe/index.d.ts (hunk 3 above)
Comment (afholderman) [reply types/apostrophe/index.d.ts]: That came in as part of the defs generated by dts-gen. Removing it did not seem to effect anything so I'll push that change.

File: types/apostrophe/index.d.ts (hunk 4 above)
Comment (afholderman) [reply types/apostrophe/index.d.ts]: Ahh thank you! I could not find the syntax for making that parameter optional!

**Review by amcasey — APPROVED** [review]
Looking good!

**Pull request merged**